src/diff/children.js

```diff
@@ -123,6 +123,10 @@ export function diffChildren(
 						firstChildDom = newDom;
 					}
 
+					if (excessDomChildren != null) {
+						excessDomChildren[excessDomChildren.indexOf(newDom)] = null;
```

Comment:
Just merged #2274 which has a similar fix. Are they the same?

Reply:
Did not notice this, you can remove the lines in diffElementChildren and use the tests in master to double check.

Reply:
merged and extracted the common `if` condition outside